### ISSUE TITLE
Pipeline filter view doesn't handle warnings well

### ISSUE THREAD

**Issue by tomspilman** [mono/MonoGame#4460]
Sometimes tools like shader compilers will dump out `warning` messages like error messages.  Here is an example of that:

https://gist.github.com/tomspilman/f502f209178e3337bc2e

The filter view in the Pipeline tool doesn't correctly detect that as the output of one build content item.

**Comment by tomspilman** [comment]
@cra0zy  @nkast ?

**Comment by nkast** [comment]
I tried to replicate that by creating a new fx and add a typo.
for example 'outputt.Position = mul(input.Position, WorldViewProjection);' 

What i got was a double escaped string!
"P:\\\\TestPipeline\\\\NewEffect.fx(28,2): error X3004: undeclared identifier 'outputt'\n"
This comes from: https://github.com/mono/MonoGame/blob/develop/Tools/2MGFX/EffectObject.hlsl.cs#L53-L57
It's not possible to parse this type of filenames, any idea of how we can handle this bug?

**Comment by tomspilman** [comment]
> I tried to replicate that by creating a new fx and add a typo.

Well a warning isn't the same as an error...  I will have to give you some specific code to trigger a warning in the HLSL compiler.

> What i got was a double escaped string!

Hum....  we get that string from the stdout/stderr streams.  I guess those streams are generating these double slashes.

> It's not possible to parse this type of filenames, any idea of how we can handle this bug?

I would say stop trying to parse it as a file name... just accept anything before the parenthesis and colon.

**Comment by nkast** [comment]
> I would say stop trying to parse it as a file name... just accept anything before the parenthesis and colon.

Yeah, we have already relaxed filename because of win/unix. This probably already works.

**Comment by cra0zy** [comment]
So problems that are present are:
 - errors and warning that fx files throw out aren't registered
 - if an error occurs it still shows fx file as success

As to what we should do for filter view, I suggest that when we parse:
```
C:\\Users\\cra0z\\Desktop\\gmjosack-xna-2d-shader-examples-07cfe1a5aafb\\ShaderTests\\ShaderTestsContent\\effects\\Bevels.fx(5,5): error X3000: unrecognized identifier 'flot4'
C:\\Users\\cra0z\\Desktop\\gmjosack-xna-2d-shader-examples-07cfe1a5aafb\\ShaderTests\\ShaderTestsContent\\effects\\Bevels.fx(5,11): error X3000: unrecognized identifier 'color'
```

we shorten it to:
```
Bevels.fx(5,5): error X3000: unrecognized identifier 'flot4'
Bevels.fx(5,11): error X3000: unrecognized identifier 'color'
```

and put that withing the filter view node children.

Also we should probably add an icon for a warning.

And the code that should allow us to check if the fx file had an error would be:
```c#
if (newline.Replace("\\\\\\\\", "/").Replace("\\\\", "/").StartsWith(previousline))
```
